cEP-0024.md

```diff
@@ -0,0 +1,226 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation in [GitMate](https://gitmate.io/) to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs opened on master of forks.
+Many of the plugins in GitMate are general plugins, which are required by every
+organization, but coala needs more plugins which support coala development
+workflow. This project adds plugins to auto-review, auto-reject, auto-label
+PRs, assign, unassign issues, squash commits.
+
+## Changes in coala workflow
+
+### Use increasing difficulty level order to assign issues
+
+Current assign process in coala is to check team membership of a user,
+difficulty level of the issue and assign issue to user if eligible. To
+implement current assign process in GitMate, IGitt should support teams,
+but it is difficult to implement, because there are no teams in GitLab.
+
+Considering teams as groups in GitLab is a possible solution for IGitt
+implementation, but requires extra sync tools to sync GitHub team and GitLab
+group memberships, which nobody has built yet.
+
+Team membership on GitHub is not publicly accessible information, so any system
+which wants to implement the current 'assign' system needs to have a master
+org key, which isn't good.
+
+Best possible solution would be using a increasing difficulty level order,
+so that a user can be assigned to an issue only if user has solved issues of
+lower difficulty level before requesting to get assigned to a higher difficulty
+level issue.
+
+#### Drawbacks
+
+If user completed the lower difficulty level issues in GitHub and want to
+work on a higher difficulty level issue in GitLab, he has to repeat the
```

Comment:
he -> they .  gender neutral.

but ...

I think what you are describing is a GitMate/IGitt limitation which is easy to fix.

GitMate/IGitt are designed to work on a per-repo level.

Your proposed model already requires that this plugin (and possibly others), work on a per-org level, with the limitation that the GitLab org isnt the same as the GitHub org.

The next leap is to link together the GitLab org and the GitHub org.  This is easy when they have the same name, as is the case with `coala` and many other orgs which are straddling GitHub and GitLab.

So you can expand/rephase this 'drawback' to explain it is possible to resolve this, but it is out of scope for your project.